### DISCUSSION TITLE
Connects to one computer but not two others. Error is "Beats me what happened to the serial port"

### DISCUSSION THREAD

**Discussion by M4TVW** [collin80/SavvyCAN#632] [Q&A]
This is a strange one. I have been trying to get my EVTVDUE working for a few days but having trouble getting it working on either of my laptops. The strange thing is it works fine on my desktop with the same version of Savvycan!

I have attached a screen shot of the error i'm getting on both laptops. 
![Screenshot 2023-05-06 203251](https://user-images.githubusercontent.com/86199399/236643462-67049a06-3cdf-435f-80c4-32dbdf31f3ba.png)

I'm running the latest GVRET that i can find, along with all the latest versions of the library's, although i get the same issue even using the compiled version from the savvycan website so not sure why it works on one computer but not two others.

On my desktop it says connected and i can see the serial reply of the DE AD tags, which looking at other threads show its working. On the two laptops it just shows "Beats me what happened to the serial port"

Any ideas? I'm at my wits end with this now.

**Comment by collin80** [comment]
Someone else recently ran into this. For some reason at least certain versions of Windows use a driver that is not compatible. The other user offered the solution:

https://github.com/collin80/SavvyCAN/discussions/630

Basically, you need the driver that the Arduino IDE installs.

**Reply by M4TVW** [reply]
Thanks Collin. That was me also. I thought I had fixed it with that but sadly not. It got me further on but still the same issue.
I installed the arduino IDE on my desktop just to make it the same as my laptop and it still works on my desktop using the arduino driver. However with everything set up exactly the same on two other laptops it does not work and just says it doesn’t know what happened to the serial port. 
Is there a power issue when powering the board from a laptop as that’s the only thing I can say is different really.

I have also tried running a Ubuntu VM on the laptop and it says the port is already in use on that so trying to work on that now. I thought buying the correct PCB would make this easy! 😂

**Reply by collin80** [reply]
Oh, sorry, yes I see now that you posted both topics... Whoops!

It's certainly strange. I would have expected it to work on all your devices or none. That it works on one and not two others is quite strange. I don't remember really anyone else reporting this issue either. You could be right about power issues. Laptops generally output a smaller voltage than desktops so it's possible that they are not providing enough power or voltage and this is interfering with the operation. Have you tried powering the board via 12v as well? This would rule out low power as being the issue.

**Reply by M4TVW** [reply]
I have just tried powering it with 12v from the battery and sadly the same issue, it didn’t make a difference. 

If I continue down the VM route, do you think I will be able to get it to work on Linux via virtual box? Or as it’s a serial port will it always be in use already via the host windows OS so it will never transfer to the VM properly?

**Comment by M4TVW** [comment]
So i managed to get it connected in the linux VM, (it was a permissions issue on the serial port). However have a look at the console on this one. 

![Screenshot 2023-05-08 125202](https://user-images.githubusercontent.com/86199399/236817544-a00f0887-31eb-41f7-a5e0-457b1605c720.png)

So it connects, managed to get data from the serial and then drops connection again with "failed to connect to GVRET".

Very strange and i cant put my finger on it. Is it right that i seem to have everything in the console twice? Connecting to GVRET twice, Write to serial twice etc. Does this point towards an issue or is that just how it is?

**Reply by collin80** [reply]
Everything happening twice is somewhat normal. There are two basic types of serial device it will try to connect to: the GVRET design based around a SAM3X chip and the ESP32 based designs. The ESP32 needs special settings to work and those settings are incompatible with the SAM3x based boards so it tries two different ways. Thus it's normal to potentially see it try twice on some boards but it shouldn't take more than twice. I see a reply from the board when it tries the standard serial mode. So, it almost seems to be working. But, that reply is a reply to one of the requests in the middle and not the first request in line which is strange. It's like parts of the traffic are being lost somehow. I can't quite explain how that could happen on a board with native USB.

One thing you might try is to use something like the Arduino IDE to bring up a serial monitor with the board plugged in and try to capture some CAN traffic. It will still be able to capture traffic without savvycan, it'll just dump the frames onto your serial console. If this works on the machines that weren't connecting to savvycan then we can point the finger at the interaction with savvycan. But, if the serial console doesn't work either then that points to  a driver issue or the board being somewhat partially faulty. It's hard to explain why it would work on one device but not others though.

**Reply by M4TVW** [reply]
So i tried the serial monitor and it seems that it won't even connect to the board via that. The error is as per the picture and you can see the board connected in my device manager. 

![Screenshot 2023-05-09 135206](https://github.com/collin80/SavvyCAN/assets/86199399/0da026b4-f83b-4d12-a8be-c0570f0e9fef)

The IDE has no problem with writing to the board and will upload a sketch fine, it's just being able to read the serial port after that.

I have just plugged it into a fresh desktop computer and it works straight away. I can see the serial output of the board (Not CAN data as i don't have a cable that long!) in the serial monitor no problems so it seems to be computer based. 

I'm looking for a way to nuke the arduino IDE including the drivers but i cannot find it. For example i just installed the IDE on the desktop and it offers to install the Arduino drivers again. I cant get to that point on the laptop.
I think i'm going to run a fresh windows install on my laptop and see if that works straight away. I can only narrow it down one by one at this point.

**Reply by collin80** [reply]
Sure sounds like a bad driver to me. Also, here is something that might help the confusion - the chip on board registers as two completely different interfaces. For programming there is a programming interface. When not programming there is a separate serial emulation interface. They are different and use different drivers. The programming interface will say something like BOSSA. You can see it if you plug the board in, hold down the ERASE button for 2 seconds, unplug, and plug back in. The way it comes up then will be the BOSSA programming port instead of the normal serial port. Either way it's a COM port but they'll probably be different numbers too. You might then wonder how it works to program it normally. Well, deep in code there's a routine that looks for someone setting the serial interface to 1200 baud. If you do that it reboots into programming mode with the programming port. So, it seems this 1200 baud shift must be happening successfully but actually opening the port doesn't. I have to suspect a bad driver still but I have no idea how some machines are getting a bad driver and some are not. The only other explanation I can offer is maybe a slightly bad cable? That seems like a real stretch as it works on some machines. It's certainly an odd situation.
